pkg/config/api_server_config.go

```diff
@@ -40,17 +41,24 @@ type APIServerConfig struct {
 	APIKeyCacheDuration time.Duration `env:"API_KEY_CACHE_DURATION,default=5m"`
 
 	// Verification Token Config
-	// Currently this does not easily support rotation. TODO(mikehelmick) - add support.
 	VerificationTokenDuration time.Duration `env:"VERIFICATION_TOKEN_DURATION,default=24h"`
 	TokenSigningKey           string        `env:"TOKEN_SIGNING_KEY,required"`
 	TokenSigningKeyID         string        `env:"TOKEN_SIGNING_KEY_ID,default=v1"`
 	TokenIssuer               string        `env:"TOKEN_ISSUER,default=diagnosis-verification-example"`
+	// In order to rotate keys, one sets these values to the 'old' key ID allowing it to still verify
```

Comment:
This backs us into a world where there can only ever be two valid keys. While I understand that's our _current_ situation, given the frequency of change in this project, I wonder if it makes sense to have an array instead where the "first" element is the newest key and all other ones are "old"

Reply:
then you have to have parallel arrays w/ kid and key ref.

The other option would be to bring this up one level where we take a cryptoKey and allow all non-destroyed versions for validation and the primary for signing.
But - I then we have to support that across all our KMS implementations

Needing more than 2 valid keys means that you're rotating twice in a 24 hour period which hopefully isn't ever necessary.

If we want this to be more flexible, I think we can push it into the database and manage it like the per-realm signing keys.

We do need to have a general 'admin' page in the UI still anyway, to allow for realm creation.

Reply:
Up to you. I just know the moment we do this, we'll need to support a third...

Reply:
will switch to arrays in the config